digitalocean/Tag.py

```diff
@@ -79,29 +80,33 @@ def __remove_resources(self, resources):
         return self.__get_resources(resources, method='DELETE')
 
 
-    def __extract_resources_from_droplets(self, data):
+    def __build_resources(self, data, object_class: str, resource_type: str):
         """
-            Private method to extract from a value, the resources.
+            Private method to build the `resources` field used to tag/untag 
+            objects.
             It will check the type of object in the array provided and build
             the right structure for the API.
+            The 2nd and 3rd arguments specify the object class and the latter 
+            takes the resource type as required by DO API.
         """
         resources = []
         if not isinstance(data, list): return data
-        for a_droplet in data:
+        for obj in data:
```

Comment:
was it really necessary to rename to a generic `obj` instead of `a_droplet`?

Reply:
> was it really necessary to rename to a generic obj instead of a_droplet?

Again, this method is now more generic than before: it can be used to build `resources` field for any resources that support it, so it does not make sense to keep `a_droplet`.

> Please if you have to rename, think about something that another maintainer can understand without the need of an IDE or a deep knowledge of the source code/api.

This is a variable inside a method, how does that influence the API?

I agree this name is not perfect, but I can't find something that's both as generic as the method is and not too generic at the same time. If you come with something better, tell me.

Reply:
> This is a variable inside a method, how does that influence the API?
One of the principle that I am trying to push in this library is to make it understandable even for a junior. So `obj` is no good :( What if we rename it into `api_generic_response`? Will this new name help a new developer? Does it make sense to you?  😅 

Thank you btw for your help 👍 😄

Reply:
> One of the principle that I am trying to push in this library is to make it understandable even for a junior on its first day of work. So obj is no good :(

Got it, sounds like a reasonable expectation actually :smile: 

> What if we rename it into api_generic_response? Will this new name help a new developer? Does it make sense?

Well, I'm not sure it's a better name :| 1. We're not manipulating responses in this method, but DTOs implemented by this lib ; 2. The name is a bit cumbersome IMHO.
To be honest, I'm not totally happy with that `resources` variable too (the one built by the method).

I came up with `obj` because `resources` was already used, but this would have been the best fit IMHO. But then we need to differentiate the input resources and the resources API field we build. What we can do to enhance both names would be to rename the current `resources` into `resources_field` and `obj` into `resource_to_tag`. WDYT?

Reply:
sounds like a great idea! 👍